### ISSUE TITLE
[bug]Variable substitutions

### ISSUE THREAD

**Issue by macthestack** [moonrepo/moon#968]
I'm trying to use env vars but I keep running into problems:

When trying to substitute in the "command" I get this problem:

```
"cr.io/${PKG_NAME}" -> cr.io/${PKG_NAME}
'cr.io/${PKG_NAME}' -> cr.io/${PKG_NAME}
cr.io/${PKG_NAME} -> cr.io/${PKG_NAME}
cr.io/ ${PKG_NAME} -> cr.io/ package_name
```

When trying to use variable substitutions it doesn't seem to pick up project level env vars.

```
PKG_NAME: "boilerplate_rust"
TAG_NAME: "cr.io/${PKG_NAME}"
```

TAG_NAME just becomes `cr.io/`

It's the same if I put TAG_NAME in task level env.

That leaves me without any options to do something like this:

`docker push cr.io/${PKG_NAME}`

or 

`docker push ${TAG_NAME}`

I expect I've just missed something but can't figure out what.

**Comment by milesj** [comment]
@macthestack What's your task config look like?

**Comment by macthestack** [comment]
Right now it's just hard-coded values similar to this:

```
env:
  PKG_NAME: "package_name"
  TAG_NAME: "cr.io/package_name"

tasks:
  docker_build:
    command: docker build --build-arg PKG_NAME -t ${TAG_NAME} .
```

I would prefer to be able to write something like this:

```
env:
  PKG_NAME: "package_name"

tasks:
  docker_build:
    command: docker build --build-arg pkg=${PKG_NAME} -t cr.io/${PKG_NAME} .
```

or:

```
env:
  PKG_NAME: "package_name"
  TAG_NAME: "cr.io/${PKG_NAME}"

tasks:
  docker_build:
    command: docker build --build-arg pkg=${PKG_NAME} -t ${TAG_NAME} .
```

or maybe even:

```
env:
  PKG_NAME: cargo metadata --format-version 1 | jq -r '.packages[] | select(.source == null) | .name'

tasks:
  docker_build:
    command: docker build --build-arg pkg=${PKG_NAME} -t cr.io/${PKG_NAME} .
```

...although the last one probably needs a separate discussion.

**Comment by milesj** [comment]
@macthestack At first glance, I think this is just how Bash works? You can test it locally.

```
$ FOO=bar echo $FOO

```

It echos a default string. But if we set the env var in another command, it works.

```
$ FOO=bar 
$ echo $FOO
bar
```

The project-level `env` vars are passed to the child process: https://doc.rust-lang.org/std/process/struct.Command.html#method.env But the parent is the one doing the expansion.

**Comment by milesj** [comment]
Landed in v1.11 https://moonrepo.dev/blog/moon-v1.11#unlocked-features
